<launch>
	<node pkg="bridge_px4" type="setpoint_publisher_node" name ="setpoint_publisher_node" output="screen">
	</node>
	
	<arg name="namespace" default="carl" /> <!--namespace of asset -->
	<arg name="asset" default="carl"/> <!-- name of vrpn asset -->
	<arg name="server" default="mocap.local"/>  <!-- vrpn server on local network  -->
	<arg name="pose_topic" default="mavros/vision_pose/pose" /> <!-- output pose stamped on local namespace  -->

	<node pkg="vrpn_client_ros" type="vrpn_tracker_node" name="vrpn_tracker_node" output="screen">
		<param name="tracker_name" value="$(arg asset)"/>
		<param name="server" value="$(arg server)"/>
		<param name="port" value="3883"/> <!-- should be 3883 -->
		<param name="update_frequency" value="100"/> <!-- maybe vrpn poll rate?, msg rate is set by mocal stream hz  -->
		<param name="use_server_time" value="false"/> <!-- set false for local robot clock stamp  -->
		<param name="broadcast_tf" value="false"/> <!-- enable tf-->
		<param name="frame_id" value="world"/> <!-- base link for tf -->
		<remap from="$(arg asset)/pose" to="$(arg pose_topic)" />
	</node>

	<include file="$(find mavros)/launch/px4.launch">
		<arg name="fcu_url" default="/dev/ttyS4:921600" />
		<arg name="gcs_url" default="" />
		<arg name="tgt_system" default="1" />
		<arg name="tgt_component" default="1" />
		<arg name="log_output" default="screen" />
		<arg name="fcu_protocol" default="v2.0" />
		<arg name="respawn_mavros" default="false" />

		<include file="$(find mavros)/launch/node.launch">
			<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
			<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="$(arg gcs_url)" />
			<arg name="tgt_system" value="$(arg tgt_system)" />
			<arg name="tgt_component" value="$(arg tgt_component)" />
			<arg name="log_output" value="$(arg log_output)" />
			<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
			<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
		</include>
	</include>
</launch>
